crates/miden-testing/src/tx_context/context.rs

```diff
@@ -91,15 +91,8 @@ impl TransactionContext {
 
         let test_lib = TransactionKernel::kernel_as_library();
 
-        let source_manager =
-            alloc::sync::Arc::new(miden_objects::assembly::DefaultSourceManager::default())
-                as alloc::sync::Arc<
-                    dyn miden_objects::assembly::SourceManager + Send + Sync + 'static,
-                >;
-
-        // TODO: SourceManager: Load source into host-owned source manager.
         // Virtual file name should be unique.
-        let virtual_source_file = source_manager.load(
+        let virtual_source_file = self.source_manager.load(
```

Comment:
This works for the usage of `execute_code`, but when using `execute_code_with_assembler` directly, one would have to make sure that the assembler passed in also uses `self.source_manager`, so maybe we should parametrize it?

Not sure that's the best way though - we can also leave as-is for this PR.

Reply:
I'd leave this as is for this PR and and add a note to the API, we'd need the acessor on `Assembler` back or use a closure for constructing an `Assembler`.